File: make_mesh.py
import numpy  as np

LonMin=2.0
LonMax=10.0
LatMin=41.0
LatMax=45.0
inc=1.5;    #in milles   (minutes of arcmax)
arx='in/nodes.npz'
#########################################3
#LonMin=14.66
#LonMax=17.83
#LatMin=36.08
#LatMax=39
#inc=1.5;    #in milles   (minutes of arcmax)
#arx='in/nodes_sn.npz'
####################################### Fi de usuari

inc=inc/60.0    # in deg


Nx=int(np.floor((LonMax-LonMin)/inc)+2)
Ny=int(np.floor((LatMax-LatMin)/inc)+2)
tira_lon=[]
for i in range(Nx):
    tira_lon.append(LonMin+i*inc)
tira_lat=[]
for j in range(Ny):  
    tira_lat.append(LatMin+j*inc)

nodes=np.zeros((Nx*Ny,2))
#print( ' Nx = {:6d} ---   Ny = {:4d}\n'.format(Nx,Ny))
print('longituds    {:8.3f}    -----   {:8.3f} \n'.format(tira_lon[0],tira_lon[-1]))
print('latituds     {:8.3f}    -----   {:8.3f} \n'.format(tira_lat[0],tira_lat[-1]))
for j in range(Ny):   
    for i in range(Nx):
        nodes[Nx*j +i,0]=tira_lon[i]
        nodes[Nx*j +i,1]=tira_lat[j]
inc=inc*60
np.savez_compressed(arx, nodes,inc, Nx,Ny,LonMin,LonMax,LatMin,LatMax,tira_lon,tira_lat)

File: Workshop SIMROUTE Academics/SIMROUTE Academics/make_mesh.m
clear all
close all

%%%%%%%% CREATES MESH:

%Mesh Dimension (degrees)

%CASE TESTS
%MED
LonMin=-5.5;LonMax=17;LatMin=35;LatMax=45; 
%BAL
%LonMin=-5;LonMax=12.9;LatMin=51;LatMax=60; 

%Resolution:
inc=3;% ALPHA; in milles   (minutes of arcmax)

%%%%%%%%%%%%% END OF INPUTS %%%%

axis([LonMin LonMax LatMin LatMax]);
%Make the mesh within these two limits and with the inc increase
%Calculate Nx i Ny
inc=inc/60.0;
Nx=floor((LonMax-LonMin)/inc)+1;
Ny=floor((LatMax-LatMin)/inc)+1;
nodes=zeros(Nx*Ny,2);
nodes(:,1)=[1:1:Nx*Ny]; 

tira_lon=[LonMin:inc:LonMin+inc*(Nx-1)];
tira_lat=[LatMin:inc:LatMin+inc*(Ny-1)];
% Build nodes: 
for j=1:Ny
  nodes(Nx*(j-1)+1:Nx*j,1)=tira_lon;
  nodes(Nx*(j-1)+1:Nx*j,2)=tira_lat(j);
end
figure(1)
hold on
plot(nodes(:,1),nodes(:,2),'.')

plot(nodes(Nx,1),nodes(Nx,2),'*r')
plot(nodes(1,1),nodes(1,2),'*r')
plot(nodes(Nx+1,1),nodes(Nx+1,2),'*g')
load in/ldc_euro_i_mask.mat;

plot(lon,lat,'r-')
inc=inc*60; % to change it into minutes
save('in/nods','nodes','Nx','Ny','inc','LonMin','LonMax','LatMin','LatMax','tira_lat','tira_lon')
disp('Mesh done')